pkg/gardenadm/cmd/token/delete/options.go

```diff
@@ -6,39 +6,73 @@ package delete
 
 import (
 	"fmt"
+	"regexp"
 	"strings"
 
 	"github.com/spf13/pflag"
+	bootstraptokenapi "k8s.io/cluster-bootstrap/token/api"
 
 	"github.com/gardener/gardener/pkg/gardenadm/cmd"
 )
 
+var (
+	validBootstrapToken       = regexp.MustCompile(`^[a-z0-9]{6}.[a-z0-9]{16}$`)
+	validBootstrapTokenID     = regexp.MustCompile(`^[a-z0-9]{6}$`)
+	validBootstrapTokenSecret = regexp.MustCompile(`^bootstrap-token-[a-z0-9]{6}$`)
+)
+
 // Options contains options for this command.
 type Options struct {
 	*cmd.Options
-	// TokenID is the ID of the token to delete.
-	TokenID string
+	// TokenValues is the name of the Secret holding the bootstrap token, or the full token of the form
+	// "[a-z0-9]{6}.[a-z0-9]{16}", or the token ID of the form "[a-z0-9]{6}" to delete.
+	TokenValues []string
+	// TokenIDs are the actual token IDs parsed from the token values.
+	TokenIDs []string
 }
 
 // ParseArgs parses the arguments to the options.
 func (o *Options) ParseArgs(args []string) error {
-	if len(args) > 0 {
-		o.TokenID = strings.TrimSpace(args[0])
+	for _, arg := range args {
+		o.TokenValues = append(o.TokenValues, strings.TrimSpace(arg))
 	}
 
 	return nil
 }
 
 // Validate validates the options.
 func (o *Options) Validate() error {
-	if o.TokenID == "" {
-		return fmt.Errorf("must provide a token ID to delete")
+	if len(o.TokenValues) == 0 {
+		return fmt.Errorf("must provide at least one token value to delete")
+	}
+
+	for _, tokenValue := range o.TokenValues {
+		switch {
+		case validBootstrapToken.Match([]byte(tokenValue)):
+		case validBootstrapTokenID.Match([]byte(tokenValue)):
+		case validBootstrapTokenSecret.Match([]byte(tokenValue)):
+		default:
+			return fmt.Errorf("invalid token value %q - accepted formats are %q or %q or %q", tokenValue, validBootstrapToken, validBootstrapTokenID, validBootstrapTokenSecret)
+		}
```

Comment:
Maybe I have too much past in languages with automatic fallthrough in switch statements, but this took some time to understand. I would prefer an `if` statement, but that may just be me.
WDYT?